flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveTableInputFormat.java

```diff
@@ -113,16 +114,74 @@ public void configure(org.apache.flink.configuration.Configuration parameters) {
 
 	@Override
 	public void open(HiveTableInputSplit split) throws IOException {
-		if (!useMapRedReader && useOrcVectorizedRead(split.getHiveTablePartition())) {
+		HiveTablePartition partition = split.getHiveTablePartition();
+		if (!useMapRedReader && useOrcVectorizedRead(partition)) {
 			this.reader = new HiveVectorizedOrcSplitReader(
 					hiveVersion, jobConf, fieldNames, fieldTypes, selectedFields, split);
+		} else if (!useMapRedReader && useParquetVectorizedRead(partition)) {
+			this.reader = new HiveVectorizedParquetSplitReader(
+					hiveVersion, jobConf, fieldNames, fieldTypes, selectedFields, split);
 		} else {
 			this.reader = new HiveMapredSplitReader(jobConf, partitionKeys, fieldTypes, selectedFields, split,
 					HiveShimLoader.loadHiveShim(hiveVersion));
 		}
 		currentReadCount = 0L;
 	}
 
+	private boolean isVectorizationSupport(LogicalType t) {
```

Comment:
I think this method actually returns whether a type is **not** supported?

Reply:
Ah... yes, I will change name to `isVectorizationUnsupported`